### DISCUSSION TITLE
Exception Coalescence

### DISCUSSION THREAD

**Discussion by Ekl1p5e** [dotnet/csharplang#6662] [Language Ideas]
I think it would be useful to add an exception coalescence operator in order to avoid using the ternary operator for a check first. Roughly, it would look something like this:

enum Color { Red, Blue, Orange, };

Color GetColorFromString(string color)
{
  switch(color)
  {
    case "Blue":
      return Color.Blue;
    default:
      throw new Exception();
  }
}

object GetObj(string color)
{
  return new
 {
    COLOR = GetColorFromString(color) *? default,
  }
}

the operator isnt important, but i think this would be useful for making concise expressions.

**Comment by DavidArno** [comment]
My concern here is that - going by your example - you appear to want to make it easier to use exceptions for control flow, rather than to handle exceptional events. Using exceptions for control flow is widely seen as an anti-pattern, so is best avoided.

Again, just focusing on your example, you could use C#'s equivalent to optional types - nullability - to convey the same information without needing exceptions:
```c#
Color? TryGetColorFromString(string color) =>
    color switch {
        "Blue" => Color.Blue,
        _ => null
    };


object GetObj(string color)
{
    return new
    {
        COLOR = TryGetColorFromString(color) ?? default
    };
}
```

**Reply by MgSam** [reply]
I think more consistent naming here would be `GetColorFromStringOrDefault()`. The `Try` methods all return booleans and the actual result in an `out` parameter.
